### ISSUE TITLE
Bug report: In Vibrate mode only silent notification when screen is off or app in background

### ISSUE THREAD

**Issue by buergi** [ncosgray/cuppa_mobile#72]
**Steps to reproduce:**
1) Set notification settings for Cuppa
![Screenshot_20220328_201014](https://user-images.githubusercontent.com/1948339/160460316-16a6037d-6899-480f-b135-5048ab6e6979.jpg)
1) Put device in non-silent mode (ringtone volume > 0)
2) Start the timer
3) Lock screen (short press of power button)
4) Wait until countdown over => Audible notification
5) Put device in vibrate only mode
6) Start the timer
7) Keep screen on, wait until countdown over => Device vibrates
8) Start timer again
9) Press home button to put app to background
10) Wait until countdown over => Silent banner notification appears, no vibration
11) Start timer again
12) Lock screen (short press of power button)
13) Wait until countdown over => Device lights up and shows lockscreen notification, no vibration

**Expected behavior:** Actually I'd expect an audible notification always (unless sound is set to mute in Notification settings), like for an alarm app. But at least I'd expect the device to vibrate when tea is ready.

I tested notifications with Tasker with identical notification settings. This works, no matter of the notification priority. So I'm not sure what is the issue here.

However, I guess I'd, anyway, recommend to use the "alarm" audio stream to play an alarm instead of relying on normal notifications (which use the "ringtone" stream I guess).

**Observed behavior:** Only a silent lock screen notification or notification banner is shown. 

**Device information:** Huawei P20 Pro, Android 10

**Comment by ncosgray** [comment]
@buergi I am not certain I'll be able to resolve your issue, since I can't reproduce it myself. On my Android 12 Pixel 3a, Cuppa alerts do not behave in the way you describe. I have tried dozens of different tests and Cuppa's notifications always produce a vibration if the OS settings allow it.

You could try uninstalling and reinstalling Cuppa, as that resets your OS notification settings for the app.

**Comment by buergi** [comment]
I tried reinstalling but no change, as long as the phone is in vibrate mode and the app is in background only a visual notification appears. It only works properly in normal none-mute ringmode.
Maybe I find some time to check the code and try to find a work around.

**Comment by ncosgray** [comment]
Okay, thanks for giving that a try.

I've been testing the app with alarm notifications instead of event notifications and I've found I actually prefer that. While I am not certain it'll fix this issue, I suspect you're right that it will at least help. Look for that change in the next release (it'll unfortunately require reinstalling again because the notification channel gets modified).

**Comment by ncosgray** [comment]
Hello again! I'm curious, now that version 1.6.0 is out -- did moving from events to alarms resolve the notification issues for you?

If you haven't already, you'll need to reinstall Cuppa again in order to update the notification channel settings for alarms. (That's an Android OS restriction.)

**Comment by buergi** [comment]
Hi there, it has gotten better in a sense that the app is able to ring while the display is off. But is has not gotten better in the sense that the app can be reliably used as a timer.
Here are two videos as a proof. I increased the speed in the boring parts to 8x.
I made both videos after installing v1.6.5 freshly from F-Droid. I then even disabled battery optimization for cuppa in the operating system's settings. Apart from that I did not change any settings.
I made both videos right after each other, in the first one there was no notification at all until I unlocked the screen.

https://user-images.githubusercontent.com/1948339/187070198-3e900be2-2210-4a8a-8395-89850357f6f9.mp4

In the second video I did exactly the same, but this time the app rang, but almost 2 minutes early, so as you can see when the timer finished at the end there was no notification or alarm, although the display was on.

https://user-images.githubusercontent.com/1948339/187070203-211d9ec1-9b64-4f22-b187-101a3e77cb25.mp4

**Comment by ncosgray** [comment]
Wow, that is truly bizarre! Are you still using the same Huawei device with Android 10? It seems as if Huawei has added more aggressive alarm/wakeup limitations compared to stock Android, presumably as a battery-saving measure.

I will certainly look at other ways of setting alarms in Cuppa (perhaps setExactAndAllowWhileIdle would work better than setExact?) but it will all be guesswork as I cannot reproduce any similar issues on my Pixel.

**Comment by buergi** [comment]
Yes, exactly Huawei P20 Pro with Android 10. Also still using the old "EMUI 10" system as when I filed the issue. Huawei just began rolling out a new "EMUI 12" which, however, is also based on Android 10, as far as I understood. I'm currently trying to postpone updating for them moment, as I've read the news version eats way more battery...

**Comment by ncosgray** [comment]
@buergi thank you so much for reporting the issue and creating these helpful videos. Thanks also for offering to test. If you have a moment to install the attached build -- which uses a different method (setExactAndAllowWhileIdle) to create notification alarms and supposedly override idle/doze mode on Android -- I would be very curious to learn if it behaves better on your phone.
[Cuppa-27.apk.zip](https://github.com/ncosgray/cuppa_mobile/files/9458914/Cuppa-27.apk.zip)

**Comment by buergi** [comment]
Sorry for having left you waited for so long. I did now try it multiple times and wasn't able to reproduce at least the early alarm. For the problem that no alarm occurs at all I found out, that setting the following option in the Android Settings is crucial `Apps > App-Start > Cuppa > Manual > Check Run in Background`.
Having this set the app works pretty reliably, however sometimes it still alarms a little to late (<15s) which I consider acceptable. This is most probably just the Android system trying to save some battery.
